docs/studio/0003-permissions.md

```diff
@@ -0,0 +1,26 @@
+# 3. permissions
+
+Date: 2024-06-27
+
+## Status
+
+Accepted
+
+## Context
+
+We need to implement a permissions model for our application. Specifically, the list of permissions required are:
+
+1. publishing of pages
+2. editing of pages
+3. adding/deletion of users
+
+Number 1/2 will only be enforced at the root folders of the site, and all sub-folders/pages will instead inherit their permissions from their root-level folder.
+For example, the site has the structure (A -> B -> C), such that A is the root level folder. Both B and C will not have permissions tied to them and will instead use the permissions linked to A.
+
+## Decision
+
+We will use role-based access control for our permissions model, and we will use [casl](https://casl.js.org/v6/en/) for the implementation of the library
```

Comment:
We should mention that there are 2 modes of permissions being supported. By default all new sites get a site-wide RBAC for publisher, editor, admin

On a opt-in basis we will allow for granular resource-level based perms on the root level folders

Reply:
good point, will do